pages/linux/rusnapshot.md

```diff
@@ -0,0 +1,20 @@
+# rusnapshot
+
+> BTRFS snapshotting utility made in rust.
+> For more information: <https://github.com/Edu4rdSHL/rusnapshot>.
+
+- Create a snapshot using a config file:
+
+`sudo rusnapshot -c {{/path/to/config.toml}} --cr`
+
+- List created snapshots:
+
+`sudo rusnapshot -c {{/path/to/config.toml}} -l`
+
+- Delete snapshot by id or the name of the snapshot:
+
+`sudo rusnapshot -c {{/path/to/config.toml}} --del --id idprovidedbyrusnapshot`
+
+- Delete all snapshots of [kind] three-hours:
```

Comment:
I don't really understand what you mean by `[kind]`. Is `three-hours` a special kind of snapshot?

Reply:
Hi, rusnapshot dev here. `kind` is just an identifier for the snapshot, you can put any string that you want. Why it exists? Well, the tool is able to perform automatic cleanups of snapshots, you can use the `--keep` option to keep only the last X snapshots, but see the following scenario: I took hourly, weekly and monthly snapshots of the same subvolumes, without an identifier like `--kind` it's almost impossible to difference between them and the tool will treat all the snapshots equally. To solve this problem you can use  `--kind hourly` for the hourly snapshots, `--kind weekly` for the weekly snapshots and `--kind montly` for the monthly snapshots, that way when the tool performs the cleanups, it's able to "see" that there are three different "kind" of snapshots and that only need to clean the ones that match the current "kind", leaving the others intact.

I hope that I have been clear, let me know if you have any additional question.

Regards,
Ed

Reply:
Maybe we should call it something like `hourly` then.  @danyeet what do you think?

Reply:
No problem with that - the idea is to show that it's a time interval so I'll change it no problem.

Reply:
Also I'd rephrase the description a little. Maybe `- Delete all hourly snapshots:`.

Reply:
Yea that would be better. Just to clarify something - the time interval convention is just an example. You could maybe make something else, like backup before update, backup before restart and etc. We just thought it would be the easiest one to explain.

Reply:
I'm inclined to prefer the first (`Delete all snapshots of kind hourly`) because `hourly` is just a name, with no meaning otherwise.  It could easily be `abc`, whereupon your suggestion @marchersimon would not make as much sense (`Delete all abc snapshots:`)

Reply:
Maybe just reformat hourly to [hourly] or {{hourly}} ? 
I mean I can see both sides, I am fine with both but abstracting it a bit might not be a bad idea :).

Reply:
> the time interval convention is just an example

Ah, I tought it was all about time intervals. In that case `Delete all snapshots of kind hourly` is more universal. However in descriptions we use `` `hourly` ``, instead of `{{hourly}}` for variables.

Reply:
Gotcha.

Reply:
I have to clarify that the `--kind` option is not just for deleting, the deletion only happens if you use the `--clean` and `--keep` but the option can be used while creating new snapshots without the need of deleting previous ones. As I said initially, it's more of an identifier than anything else.